nbgrader/utils.py

```diff
@@ -61,6 +61,18 @@ def is_locked(cell: NotebookNode) -> bool:
     else:
         return cell.metadata['nbgrader'].get('locked', False)
 
+
+def has_failed(cell: NotebookNode) -> bool:
+    """Returns True if the cell contains an output indicative of an error"""
+    if not cell.cell_type == 'code':
+        raise ValueError("cell is not a code cell, so it can't fail")
```

Comment:
```suggestion
        return False
```
Maybe we don't need to raise here